src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/api/HandIntegrationTest.java

```diff
@@ -172,4 +170,135 @@ void return_correct_turnActive_for_playing_card() {
+            .build();
+
+    gameRepository.saveAll(List.of(game));
+
+    String uri = "games/" + game.getId();
+
+    webTestClient
+        .get()
+        .uri(uri)
+        .header(HttpHeaders.COOKIE, "JSESSIONID=%s".formatted(sessionId))
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectBody()
+        .jsonPath("_embedded.matches[0].hands[0].numberOfWonTricks")
+        .isEqualTo(0)
+        .jsonPath("_embedded.matches[0].hands[0].points")
+        .value(nullValue())
+        .jsonPath("_embedded.matches[0].hands[1].numberOfWonTricks")
+        .isEqualTo(2)
+        .jsonPath("_embedded.matches[0].hands[1].points")
+        .value(nullValue());
+  }
+
+  @Test
+  void return_correct_number_of_won_tricks_and_points() {
+    HttpHeaders responseHeaders =
+        webTestClient
+            .post()
+            .uri("/players")
+            .body(BodyInserters.fromValue(PLAYER_1))
+            .exchange()
+            .expectStatus()
+            .isCreated()
+            .expectBody()
+            .returnResult()
+            .getResponseHeaders();
+
+    String sessionId = getSessionIdOf(responseHeaders);
+    Player createdPlayer = playerRepository.findAll().iterator().next();
+
+    Game game =
+        GameBuilder.builder("test", gameRepository, participationRepository, playerRepository)
+            .withParticipationWith(createdPlayer)
+            .withParticipation(PLAYER_NAME_2)
+            .withMatch()
+            .withHandForPlayer(PLAYER_NAME_1)
+            .withCards(ACE_OF_CLUBS, SEVEN_OF_CLUBS)
+            .withAnnouncedScore(1)
+            .finishHand()
+            .withHandForPlayer(PLAYER_NAME_2)
+            .withCards(QUEEN_OF_CLUBS, ACE_OF_SPADES)
+            .withAnnouncedScore(2)
+            .finishHand()
+            .withRound()
+            .withPlayedCard(PLAYER_NAME_1, ACE_OF_CLUBS)
+            .withPlayedCard(PLAYER_NAME_2, ACE_OF_SPADES)
+            .finishRound()
+            .withRound()
+            .withPlayedCard(PLAYER_NAME_1, SEVEN_OF_CLUBS)
+            .withPlayedCard(PLAYER_NAME_2, QUEEN_OF_CLUBS)
+            .finishRound()
+            .finishMatch()
+            .build();
+
+    game.getLastMatch()
+        .flatMap(Match::getLastRound)
+        .ifPresent(lastRound -> lastRound.setRoundNumber(9));
+    gameRepository.saveAll(List.of(game));
+
+    String uri = "games/" + game.getId();
+
+    webTestClient
+        .get()
+        .uri(uri)
+        .header(HttpHeaders.COOKIE, "JSESSIONID=%s".formatted(sessionId))
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectBody()
+        .jsonPath("_embedded.matches[0].hands[0].numberOfWonTricks")
+        .isEqualTo(0)
+        .jsonPath("_embedded.matches[0].hands[0].points")
+        .isEqualTo(-1)
+        .jsonPath("_embedded.matches[0].hands[1].numberOfWonTricks")
+        .isEqualTo(2)
+        .jsonPath("_embedded.matches[0].hands[1].points")
+        .isEqualTo(4);
+  }
```

Comment:
Hast du absichtlich nur Tests gemacht, die gestackte Runden haben (und keine mit zwei normalen Runden)?

Reply:
Bei den punkten wird das stacken momentan nicht berücksichtigt.
(Weil wir die logik mit dem nachziehen noch nicht haben)
Es gewinnt die letzte höchste karte

Reply:
Git es in HandScoreTest